src/herder/Upgrades.cpp

```diff
@@ -318,4 +328,237 @@ Upgrades::timeForUpgrade(uint64_t time) const
 {
     return mParams.mUpgradeTime <= VirtualClock::from_time_t(time);
 }
+
+void
+Upgrades::dropAll(Database& db)
+{
+    db.getSession() << "DROP TABLE IF EXISTS upgradehistory";
+    db.getSession() << "CREATE TABLE upgradehistory ("
+                       "ledgerseq    INT NOT NULL CHECK (ledgerseq >= 0), "
+                       "upgradeindex INT NOT NULL, "
+                       "upgrade      TEXT NOT NULL, "
+                       "changes      TEXT NOT NULL, "
+                       "PRIMARY KEY (ledgerseq, upgradeindex)"
+                       ")";
```

Comment:
you're missing:
* the SQL schema upgrade code (it's in a different commit) and updates to `db-schema.md`
* maintenance implementation (see `LedgerManagerImpl::deleteOldEntries`)
* an index on `ledgerseq` (for maintenance to work)